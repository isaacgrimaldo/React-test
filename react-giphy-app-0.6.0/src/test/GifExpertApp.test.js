import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('Pruebas de componente GifExpertApp.js', () => {
    
    const wrapper = shallow( <GifExpertApp /> );

    test('Debe de mostrase correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe renderizar las defaultCategorys correctamente ', () => {
       const category = ['Goku', 'Bad bunny'];
       const wrapper = shallow( <GifExpertApp defaultCategorys ={category} />)
        
       expect(wrapper.find('AddCategory').exists()).toBe(true);
       expect(wrapper.find('GifGrid').length).toBe(category.length);
    });  
})


