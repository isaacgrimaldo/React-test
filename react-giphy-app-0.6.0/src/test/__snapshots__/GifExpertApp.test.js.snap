// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente GifExpertApp.js Debe de mostrase correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
